client/ayon_core/hosts/max/plugins/publish/validate_mesh_has_uv.py

```diff
@@ -0,0 +1,66 @@
+
+import pyblish.api
+from ayon_core.hosts.max.api.action import SelectInvalidAction
+from ayon_core.pipeline.publish import (
+    ValidateMeshOrder,
+    OptionalPyblishPluginMixin,
+    PublishValidationError
+)
+from pymxs import runtime as rt
+
+
+class ValidateMeshHasUVs(pyblish.api.InstancePlugin,
+                         OptionalPyblishPluginMixin):
+
+    """Validate the current mesh has UVs.
+
+    This validator only checks if the mesh has UVs but not
+    whether all the individual faces of the mesh have UVs.
+
+    It validates whether the current mesh has texture vertices.
+    If the mesh does not have texture vertices, it does not
+    have UVs in Max.
+
+    """
+
+    order = ValidateMeshOrder
+    hosts = ['max']
+    families = ['model']
+    label = 'Validate Mesh Has UVs'
+    actions = [SelectInvalidAction]
+    optional = True
+
+
+    @classmethod
+    def get_invalid(cls, instance):
+        invalid_mesh_type = [member for member in instance.data["members"]
+                             if not rt.isProperty(member, "mesh")]
+        if invalid_mesh_type:
+            cls.log.error("Non-mesh type objects detected")
+            return invalid_mesh_type
```

Comment:
Honestly - it's not up to this validator to check for only mesh types in model instances. It clutters the validator and doesn't sound like what it actually should be validating. It checks for uvs on meshes. Not whether everything is a mesh.

I'd say this validator should only validate meshes, ignore the rest silently.
Then if model instances may only have meshes - then that should be a separate validator.